gluten-substrait/src/main/scala/org/apache/hadoop/fs/viewfs/ViewFileSystemUtils.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.fs.viewfs
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+
+object ViewFileSystemUtils {
+
+  /**
+   * Convert the viewfs path to hdfs path. Similar to ViewFileSystem.resolvePath, but does not make
+   * RPC calls.
+   */
+  def convertViewfsToHdfs(f: String, hadoopConfig: Configuration): String = {
+    val path = new Path(f)
+    FileSystem.get(path.toUri, hadoopConfig) match {
+      case vfs: ViewFileSystem =>
+        val fsStateField = vfs.getClass.getDeclaredField("fsState")
+        fsStateField.setAccessible(true)
+        val fsState = fsStateField.get(vfs).asInstanceOf[InodeTree[FileSystem]]
+        val res = fsState.resolve(f, true)
+        if (res.isInternalDir) {
+          f
+        } else {
+          Path.mergePaths(new Path(res.targetFileSystem.getUri), res.remainingPath).toString
+        }
```

Comment:
If we don't use Java reflection, we need to add this method in ViewFileSystem:
```java
  public String convertViewfsToHdfs(String f) throws IOException {
    final InodeTree.ResolveResult<FileSystem> res;
    res = fsState.resolve(f, true);
    if (res.isInternalDir()) {
      return f;
    } else {
      return Path.mergePaths(new Path(res.targetFileSystem.getUri()), res.remainingPath).toString();
    }
  }
```

Reply:
CC @JkSelf for her comments

Reply:
@wangyum It make sense to use Java reflection instead of modifying the code in Hadoop ViewFileSystem.

> return Path.mergePaths(new Path(res.targetFileSystem.getUri()), res.remainingPath).toString();\

Can we directly return the `new Path(res.targetFileSystem.getUri())`? It seems the code here https://github.com/apache/incubator-gluten/pull/8331/files#diff-68925b6a5ee67a942a976971ae6727a45a0b166d200d8d983454169e7c3bd1f6R124 only retrieves the hdfs prefix not the full hdfs path? Correct me if wrong understand. Thanks.

Reply:
The origin path is:
```
viewfs://my-cluser/path/to/table/dt=20241209/hour=03/ns=ns7/part-01996-12a9e188-d81f-4b6b-9054-cef61acb8b44.c000.snappy.parquet
```
The `f` is:
```
viewfs://my-cluser/path/to/table/dt=20241209/hour=03/ns=ns7
```
the `res.targetFileSystem.getUri()` is:
```
hdfs://my-cluser-router
```
So we need to append remainingPath:
```
hdfs://my-cluser-router/path/to/table/dt=20241209/hour=03/ns=ns7
```

Reply:
@wangyum I see. Thanks for your explanation.